### ISSUE TITLE
roachperf: regression around 2022-09-09 [Go 1.19]

### ISSUE THREAD

**Issue by erikgrinaker** [cockroachdb/cockroach#88038]
Most pronounced on [`kv95/enc=false/nodes=3/cpu=96`](https://roachperf.crdb.dev/?filter=&view=kv95%2Fenc%3Dfalse%2Fnodes%3D3%2Fcpu%3D96&tab=gce), but also shows up on many other benchmarks.

<img width="785" alt="Screenshot 2022-09-16 at 12 43 08" src="https://user-images.githubusercontent.com/644420/190621935-c19b7ce9-4fd3-411e-8e06-bd25d4764095.png">

Jira issue: CRDB-19655

**Comment by erikgrinaker** [comment]
Reassigning to repl until we find an owner to investigate the Go 1.19 regression.

**Comment by ajwerner** [comment]
It'd be good to get CPU profiles before and after the upgrade for some workload to compare.

**Comment by erikgrinaker** [comment]
Grabbed a couple of profiles, nothing stands out immediately.

* [cpu-go1.18.pprof.gz](https://github.com/cockroachdb/cockroach/files/9601610/cpu-go1.18.pprof.gz)
* [cpu-go1.19.pprof.gz](https://github.com/cockroachdb/cockroach/files/9601613/cpu-go1.19.pprof.gz)

The profile diff shows some changes in syscalls, stack handling, and scheduling:

```
$ go tool pprof -diff_base cpu-go1.18.pprof.gz cpu-go1.19.pprof.gz 
Duration: 10.23s, Total samples = 123.19s (1204.51%)
Entering interactive mode (type "help" for commands, "o" for options)
(pprof) top10
Showing nodes accounting for -0.64s, 0.52% of 123.19s total
Dropped 1433 nodes (cum <= 0.62s)
Showing top 10 nodes out of 2221
      flat  flat%   sum%        cum   cum%
    -7.18s  5.83%  5.83%     -0.79s  0.64%  syscall.Syscall
     6.51s  5.28%  0.54%      6.51s  5.28%  runtime/internal/syscall.Syscall6
     0.63s  0.51% 0.032%      0.62s   0.5%  runtime.adjustpointers
    -0.52s  0.42%  0.45%     -0.51s  0.41%  runtime.heapBitsSetType
    -0.50s  0.41%  0.86%     -0.67s  0.54%  github.com/cockroachdb/pebble/internal/arenaskl.(*Skiplist).findSpliceForLevel
     0.43s  0.35%  0.51%      2.09s  1.70%  runtime.adjustframe
     0.40s  0.32%  0.19%      0.74s   0.6%  runtime.getStackMap
    -0.39s  0.32%   0.5%     -0.39s  0.32%  runtime.duffcopy
    -0.38s  0.31%  0.81%     -0.38s  0.31%  sync.(*RWMutex).RLock
     0.36s  0.29%  0.52%      0.38s  0.31%  runtime.(*itabTableType).find

(pprof) top10 -cum
Showing nodes accounting for 6.73s, 5.46% of 123.19s total
Dropped 1433 nodes (cum <= 0.62s)
Showing top 10 nodes out of 2221
      flat  flat%   sum%        cum   cum%
     0.04s 0.032% 0.032%      6.55s  5.32%  syscall.RawSyscall6
     6.51s  5.28%  5.32%      6.51s  5.28%  runtime/internal/syscall.Syscall6
     0.21s  0.17%  5.49%      5.17s  4.20%  runtime.findRunnable
     0.01s 0.0081%  5.50%      4.90s  3.98%  github.com/cockroachdb/pebble.(*Iterator).constructPointIter.func1
     0.04s 0.032%  5.53%      4.72s  3.83%  github.com/cockroachdb/pebble.(*Iterator).Clone
     0.02s 0.016%  5.54%      4.70s  3.82%  github.com/cockroachdb/pebble.finishInitializingIter
    -0.05s 0.041%  5.50%      4.67s  3.79%  github.com/cockroachdb/pebble.(*Iterator).constructPointIter
         0     0%  5.50%      4.64s  3.77%  github.com/cockroachdb/cockroach/pkg/storage.newMVCCIterator
         0     0%  5.50%      4.61s  3.74%  github.com/cockroachdb/cockroach/pkg/storage.(*intentInterleavingReader).NewMVCCIterator
    -0.05s 0.041%  5.46%      4.61s  3.74%  github.com/cockroachdb/cockroach/pkg/storage.(*pebbleIterator).initReuseOrCreate
```

**Comment by erikgrinaker** [comment]
For posterity, the Go 1.19 release notes say:

>The runtime will now allocate initial goroutine stacks based on the historic average stack usage of goroutines. This avoids some of the early stack growth and copying needed in the average case in exchange for at most 2x wasted space on below-average goroutines.

We disabled this behavior with `GODEBUG=adaptivestackstart=0`, which had no effect on Go 1.19 performance.

**Comment by ajwerner** [comment]
Was this profile from `GODEBUG=adaptivestackstart=0` or no?

**Comment by erikgrinaker** [comment]
No, vanilla run.

**Comment by ajwerner** [comment]
This does seem to have something to do with stack growth. 

<img width="1697" alt="Screen Shot 2022-09-19 at 5 05 47 PM" src="https://user-images.githubusercontent.com/1839234/191117167-240e2fb3-88a1-4bdc-a9ed-b705aa33a39d.png">

This is go1.19, and in go1.18:
 
<img width="1788" alt="Screen Shot 2022-09-19 at 5 10 13 PM" src="https://user-images.githubusercontent.com/1839234/191117143-75c09be3-c24a-4a97-8764-cb2da4364cca.png">

**Comment by ajwerner** [comment]
What's interesting is that the grow codec is being used in both. I don't really understand. We should try doubling the `growstack` size and see what it does for us.

**Comment by erikgrinaker** [comment]
Nice catch @ajwerner. Ran with #88187 on top of 2675c7c048 and compared with its parent on `kv95/enc=false/nodes=3/cpu=32`, and this closes the gap. In fact, 1.19 was now 1% faster, but with n=1 that doesn't really say much. Thanks!

**Comment by erikgrinaker** [comment]
Resolved by #88187.
